sfge/src/main/java/com/salesforce/logging/LoggerUtil.java

```diff
@@ -0,0 +1,51 @@
+package com.salesforce.logging;
+
+import com.google.gson.Gson;
+import com.salesforce.graph.vertex.BaseSFVertex;
+import com.salesforce.messaging.CliMessager;
+import com.salesforce.messaging.EventKey;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.Logger;
+
+public final class LoggerUtil {
+
+    public static void warnAndPostTelemetry(Logger logger, String msg, BaseSFVertex vertex) {
+        TelemetryData td = new TelemetryData(msg, EventType.LOG_WARNING, vertex);
+        CliMessager.postMessage("TelemetryData", EventKey.INFO_TELEMETRY, new Gson().toJson(td));
+        if (logger.isWarnEnabled()) {
+            logger.warn(msg);
+        }
+    }
+
+    /** An object that can be used as the base for an SFDX telemetry event. */
+    private static class TelemetryData {
+        /** Necessary property for telemetry objects. */
+        private final String eventName = "SFGE_TELEMETRY";
+
+        private final String message;
+        private final EventType eventType;
+        private final String vertexString;
+        private final String stackTrace;
+
+        public TelemetryData(String message, EventType eventType, BaseSFVertex vertex) {
+            this.message = message;
+            this.eventType = eventType;
+            this.vertexString = vertex.toMinimalString();
+            StackTraceElement[] stackTraceElements = Thread.currentThread().getStackTrace();
+            this.stackTrace =
+                    Arrays.stream(stackTraceElements)
+                            .limit(5)
+                            .map(StackTraceElement::toString)
+                            .collect(Collectors.joining("\n"));
+        }
+    }
+
+    private enum EventType {
+        LOG_WARNING,
+        PATH_ERROR,
```

Comment:
Would PATH_ERROR be the same as a Rule execution error? Or do we need something different?

Reply:
The categories are:
- Log warning: We logged a warning and kept going
- Path error(?): We encountered an unsupported case while evaluating a path, and this path can't be evaluated any further.
- Fatal error(?): Something happened that tanked the entire process, on par with an OOM error.
I'm open to different names, those were just the first ones that came to mind.